### ISSUE TITLE
Filter by country (code)

### ISSUE THREAD

**Issue by Aleksio** [komoot/photon#175]
I'm trying to filter the search results by country code, but can't seem to do so.
The readme states I can use the nominatim source as reference, and that includes country_code but I can't get it to work.

Is it actually supported? Or is the query incorrect?

http://photon.komoot.de/api?q=berlin&osm_tag=country_code:DE
http://photon.komoot.de/api?q=berlin&osm_tag=country:DE

**Comment by christophlingg** [comment]
hi @Aleksio, it is not possible to query any kind of osm tag, just the one that got considered by nominatim. For example, this query works: http://photon.komoot.de/api?q=berlin&osm_tag=place:city

Can you see the `osm_tag` and `osm_value` field of the response. That are the attributes where the filter apply.

**Comment by Aleksio** [comment]
Thanks for the quick reply, I merely looked at the list provided here:
https://github.com/openstreetmap/osm2pgsql/blob/master/output-gazetteer.cpp#L81

Is it in any way possible to filter results by country(code), so we can provide means to our users to get more accurate results? Or is such feature planned in the future?

**Comment by christophlingg** [comment]
we haven't planned to implement such a feature yet. do you always want german results only? you could simple delete all non german items from the index, queries can only return german items then.

**Comment by Aleksio** [comment]
Unfortunately not, that was only for example. The usage is international, so being able to filter by country code would be ideal.

**Comment by amenzhinsky** [comment]
I would be nice to have this feature done

**Comment by 1tft** [comment]
Hello, does anyone has a quick elastic query for deleting whole countries from elastic photon indice?

**Comment by 1tft** [comment]
I used following query and after that fired also
`curl -XPUT 'http://localhost:9200/photon/_settings?preserve_existing=true' -d '{ index.merge.policy.expunge_deletes_allowed" : "0" }'`
and
`curl -X POST "localhost:9200/photon/_forcemerge?only_expunge_deletes"`
to purge index.

`curl -X POST "http://localhost:9200/photon/place/_delete_by_query?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "bool": {
      "must": {
        "match": {
         "collector.en.ngrams": {
            "query": "Afghanistan, Albania, Algeria",
	    "operator": "or",
            "analyzer": "index_ngram"
          }
        }
      }
    }
  }
}'`

**Comment by expressIndustries** [comment]
How would I delete all elastic photon indices EXCEPT one country?  IE:  everything BUT Canada ?

**Comment by 1tft** [comment]
@expressIndustries Try to us instead of "must" a "must_not" delete query.

**Comment by expressIndustries** [comment]
Ok. thanks.  Working on getting a USA only photon database up and running with TIGER street addresses...but having trouble getting TIGER data to load into Nominatim.  Nothing happens when I run the import command: no errors are thrown and no new tables or indexes are created in nominatim database.  Stumped at the moment.

Curious if loading the graphopper DB and then removing all other countries except USA using Elasticsearch query would be similar result as trying to load all US States from Geographik and then load TIGER.  Any idea?

**Comment by lonvia** [comment]
Your comment is off-topic for this ticket. For questions about installation and use of photon please use [the mailing list])(https://lists.openstreetmap.org/listinfo/photon).

**Comment by verybigelephants** [comment]
this would be truly wonderful to have such option

**Comment by choesterle** [comment]
Where is the accurate list for the supported **tags and values**? The one provided in the readme.md (https://github.com/openstreetmap/osm2pgsql/blob/master/output-gazetteer.cpp#L81) doesn't seem to work anymore

**Comment by wottpal** [comment]
👍

**Comment by mickey-connectit** [comment]
> Where is the accurate list for the supported **tags and values**? The one provided in the readme.md (https://github.com/openstreetmap/osm2pgsql/blob/master/output-gazetteer.cpp#L81) doesn't seem to work anymore

I'm finding it similarly difficult to figure out what tags and values are supported. I would also like to filter the results by country code (or even by the `state` key on the returned JSON object) but it doesn't seem that this is possible.

**Comment by alexboss** [comment]
Indeed, would be a very helpful feature. Hope it will be implemented someday. Thanks for the Photon project anyway.

**Comment by lebadapetru** [comment]
Thank you for this package, i use the api for mvps that use some sort of geolocation features and i came across the need to filter by country as well. Will this feature every be added?
